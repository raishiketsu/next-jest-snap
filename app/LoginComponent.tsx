"use client"
import { ChangeEvent, FormEvent, useState } from 'react';

const LoginComponent = () => {
  const [username, setUsername] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const [showUser, setShowUser] = useState(false);

  const onSubmit = (event: FormEvent) => {
    setShowUser(false);
    event.preventDefault();
    console.log({ username, password });
    if (validate(username)) {
        setShowUser(true);
        return;
      }
  };

  const handleUsernameChange = (event: ChangeEvent<HTMLInputElement>) => {
    setUsername(event.target.value);
  };

  const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
    setPassword(event.target.value);
  };

  return (
    <form onSubmit={onSubmit}>
      <div>
        <label>ユーザー名</label>
        <input 
          name="username"
          placeholder="username"
          value={username}
          onChange={handleUsernameChange}
        />
      </div>

      <div>
        <label>パスワード</label>
        <input 
          name="password" 
          type="password"
          placeholder="password"
          value={password}
          onChange={handlePasswordChange}
        />
      </div>

      {showUser && (
          <p>
            送信成功
          </p>
        )}

      <button data-testid="submit" type="submit">ログイン</button>
    </form>
  );
}

export const validate = (username: string) => {
    if (username === null) {
        return false;
      }
      return true;
  };

export default LoginComponent;
