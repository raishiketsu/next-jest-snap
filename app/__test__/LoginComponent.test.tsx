import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import LoginComponent from '../LoginComponent';

describe('LoginComponent', () => {
    it('renders correctly', () => {
      const { asFragment } = render(<LoginComponent />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly', async() => {
        render(<LoginComponent />);
        const buttonElementList = await screen.findAllByRole("button");
        expect(buttonElementList).toHaveLength(1);
    });

    test("should be able to submit the form", async () => {
        render(<LoginComponent />);
        const submitButton = screen.getByTestId("submit");
        const email = screen.getByPlaceholderText("username");
        const password = screen.getByPlaceholderText("password");
    
        fireEvent.change(email, { target: { value: "test@gmail.com" } });
        fireEvent.change(password, { target: { value: "12345" } });
    
        fireEvent.click(submitButton);
    
        await waitFor(() => {
            expect(screen.getByText('送信成功')).toBeInTheDocument();
          });
      });
  });