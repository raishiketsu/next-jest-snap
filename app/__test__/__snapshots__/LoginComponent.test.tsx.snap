// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label>
        ユーザー名
      </label>
      <input
        name="username"
        placeholder="username"
        value=""
      />
    </div>
    <div>
      <label>
        パスワード
      </label>
      <input
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <button
      data-testid="submit"
      type="submit"
    >
      ログイン
    </button>
  </form>
</DocumentFragment>
`;
